kong/clustering/rpc/concentrator.lua

```diff
@@ -116,21 +180,16 @@ function _M:_event_loop(lconn)
       if n.channel == rpc_resp_channel_name then
         -- an response for a previous RPC call we asked for
         local payload = cjson_decode(n.payload)
-        assert(payload.jsonrpc == jsonrpc.VERSION)
 
-        -- response
-        local cb = self.interest[payload.id]
-        self.interest[payload.id] = nil -- edge trigger only once
-
-        if cb then
-          local res, err = cb(payload)
-          if not res then
-            ngx_log(ngx_WARN, "[rpc] concentrator response interest handler failed: id: ",
-                    payload.id, ", err: ", err)
-          end
+        if not isarray(payload) then
```

Comment:
It could be something like "[]" and it's considered an error: https://www.jsonrpc.org/specification#overview:~:text=rpc%20call%20with%20an%20empty%20Array

We need more strict check

Reply:
done, thanks.